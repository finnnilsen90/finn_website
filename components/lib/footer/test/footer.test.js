const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Footer = require('../footer.js');
const json = require('../../../../data_files/base_menu.json')

describe('component connection', () => {

    it('should be defined', () => {
        expect(Footer).toBeDefined();
    })

})  
    
describe('<Footer />', () => {

    const wrapper = mount(<Footer menu={json}/>);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });

})

describe('menu buttons from hamburger', () => {

    const wrapper = shallow(<Footer />);

    let links = {
                login: {
                    link: '/login',
                    className: 'link_footer',
                    text: 'Login'
                },
                home_page: {
                    link: '/home_page',
                    className: 'link_footer',
                    text: 'Home Page'
                },
                resume: {
                    link: '/resume',
                    className: 'link_footer',
                    text: 'Resume'
                },
                projects: {
                    link: '/home_page',
                    className: 'link_footer',
                    text: 'Projects'
                },
                contact: {
                    link: '/contact',
                    className: 'link_footer',
                    text: 'Contact'
                }
            }

    it('shold display hamburger menu object', () =>{
        expect(links).toEqual(wrapper.state().links)
    })
})

describe('menu length', ()=> {

    const wrapper = mount(<Footer />, { attachTo: document.body });

    let test_links = {
        login: {
            link: '/login',
            className: 'link_footer',
            text: 'Login'
        },
        home_page: {
            link: '/home_page',
            className: 'link_footer',
            text: 'Home Page'
        },
        resume: {
            link: '/resume',
            className: 'link_footer',
            text: 'Resume'
        },
        projects: {
            link: '/home_page',
            className: 'link_footer',
            text: 'Projects'
        },
        contact: {
            link: '/contact',
            className: 'link_footer',
            text: 'Contact'
        },
        test: {
            link: '/login',
            className: 'link_footer',
            text: 'Test'
        }
    }

    let test_result = [<li className="link_footer link_footer"><a href="/login">Login</a></li>, <li className="link_footer link_footer"><a href="/home_page">Home Page</a></li>, <li className="link_footer link_footer"><a href="/resume">Resume</a></li>, <li className="link_footer link_footer"><a href="/home_page">Projects</a></li>, <li className="link_footer link_footer"><a href="/contact">Contact</a></li>]

    it('should not be greater than 5 links', () => {
        expect(wrapper.instance().links_compile(test_links)).toEqual(test_result)
    })

    let test_result_two = [<li className="link_footer link_footer"><a href="/login">Test</a></li>];

    it('should render last part', ()=> {
        expect(wrapper.instance().links_comp_two(test_links)).toEqual(test_result_two);
    })
})

describe('column two', () => {

    let test_links = {
        login: {
            link: '/login',
            className: 'link_footer',
            text: 'Login'
        },
        home_page: {
            link: '/home_page',
            className: 'link_footer',
            text: 'Home Page'
        },
        resume: {
            link: '/resume',
            className: 'link_footer',
            text: 'Resume'
        },
        projects: {
            link: '/home_page',
            className: 'link_footer',
            text: 'Projects'
        },
        contact: {
            link: '/contact',
            className: 'link_footer',
            text: 'Contact'
        },
        test: {
            link: '/login',
            className: 'link_footer',
            text: 'Test'
        }
    }

    const wrapper = mount(<Footer menu={test_links}/>, { attachTo: document.body });

    it('should render nav_two class', () => {
        expect(wrapper).toMatchSnapshot()
    })
})