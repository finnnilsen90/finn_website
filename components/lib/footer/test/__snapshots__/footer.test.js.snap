// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders components 1`] = `
<Footer
  menu={
    Object {
      "admin": Object {
        "contact": Object {
          "className": "link",
          "link": "/contact",
          "text": "Contact",
        },
        "create_login": Object {
          "className": "create",
          "link": "/create_login",
          "text": "Create Login",
        },
        "form_submit": Object {
          "className": "form_submit",
          "link": "/form",
          "text": "Form",
        },
        "home_page": Object {
          "className": "link",
          "link": "/home_page",
          "text": "Home Page",
        },
        "logout": Object {
          "className": "link",
          "link": "/logout",
          "text": "Logout",
        },
        "projects": Object {
          "className": "link",
          "link": "/home_page",
          "text": "Projects",
        },
        "resume": Object {
          "className": "link",
          "link": "/resume",
          "text": "Resume",
        },
      },
      "logedin": Object {
        "contact": Object {
          "className": "link",
          "link": "/contact",
          "text": "Contact",
        },
        "home_page": Object {
          "className": "link",
          "link": "/home_page",
          "text": "Home Page",
        },
        "logout": Object {
          "className": "link",
          "link": "/logout",
          "text": "Logout",
        },
        "projects": Object {
          "className": "link",
          "link": "/home_page",
          "text": "Projects",
        },
        "resume": Object {
          "className": "link",
          "link": "/resume",
          "text": "Resume",
        },
      },
    }
  }
>
  <div
    className="footer_Container font"
  >
    <div
      className="nav"
    >
      <ul>
        <li
          className="undefined link_footer"
        >
          <a />
        </li>
        <li
          className="undefined link_footer"
        >
          <a />
        </li>
      </ul>
    </div>
    <div
      className="nav nav_two"
    >
      <ul />
    </div>
  </div>
</Footer>
`;

exports[`column two should render nav_two class 1`] = `null`;
